frontend/src/utils/date-time.ts

```diff
@@ -22,16 +26,78 @@ export function formatDurationFromSeconds(sec: number) {
   return formatDurationToUnit(sec, ['second', 'seconds']);
 }
 
+/**
+ * return an object containing the days, hours, minutes, and seconds
+ * equal to an input number of seconds
+ * @param sec number of seconds input
+ * @returns {days, hours, minutes, seconds} object with date data
+ */
+export function daysHoursMonthsSeconds(sec: number):
+{ days: number, hours: number, minutes: number, seconds: number } {
+  const days = Math.floor(sec/(SECONDS_IN_DAY));
+  const hours = Math.floor(sec/(SECONDS_IN_HOUR) % 24);
+  const minutes = Math.floor(sec/(SECONDS_IN_MINUTE)) % 60;
+  const seconds = sec % 60;
+  return { days, hours, minutes, seconds };
+}
+
+/**
+ * get a formatted string equal to the amount of time given
+ * an input number of seconds
+ * @param sec input number of seconds
+ * @returns string in format of '99d 11h 59m 59s'
+ */
 export function secondsToDDHHMMSS (sec: number) {
-  const days = Math.floor(sec/(60*60*24));
-  const hours = Math.floor(sec/(60*60)%24);
-  const minutes = Math.floor(sec/(60))%60;
-  const seconds = sec%60;
+  const {days, hours, minutes, seconds} = daysHoursMonthsSeconds(sec);
 
   return `${days !== 0 && days.toFixed(0) + 'd ' || ''}` + `${hours !== 0 && ('0' + hours).slice(-2) + 'h ' || ''}` +
     `${minutes !== 0 &&('0' + minutes).slice(-2) + 'm ' || ''}` + `${seconds !== 0 && ('0' + seconds).slice(-2) + 's' || ''}`;
 }
 
+/**
+ * get a string value of the relative timespan based on
+ * leading specificity, i.e. 8 months and 15 mins would output
+ * just 8 months.
+ * @param sec input number of seconds
+ * @returns string in format of `${value} ${span-of-time}`.
+ * ex: '4 months' or '8 days'
+ */
+export function secondsToRelativeAge (sec: number) {
```

Comment:
Is it possible to use translations here?
If it's difficult to achieve, ignore this comment.